### PR TITLE
fix(core,developer): variable/marker substitution in sets and strings🙀

### PR DESCRIPTION
**Pull request by srl295**
Bug: When sets contain markers, the 'map' operation fails, because the matched string was not found in the KMX+ set list.

Cause: inconsistent escaping due to the in-memory set variable being used for regex updating AND also used by core for the runtime match.

Similar for string variables (#11037)

Solution: 
1. The in-memory variables should be string literals, not regex (see #11037)
2. Need to push all escaping down to the regex substitution layer. Don't perform it 'early' in the lists.

Fixes: #11045
Fixes: #11037 

@keymanapp-test-bot skip

### CODE REVIEW

**Comment by srl295** [comment]
https://github.com/keymanapp/keyman/pull/11059/commits/82343d7973ffa003d6eaea3cd17b5e1e9c4bf2eb unrelated and may split out.

**Comment by srl295** [comment]
ldml_test still fails here.

**Comment by mcdurdin** [comment]
can you update title to make it more explicit about what it is fixing (words 'fix' and 'issues' are redundant)?

File: common/web/types/src/util/util.ts [hunk 1]
```diff
@@ -127,7 +127,7 @@ export function escapeRegexChar(ch: string) {
 }
 
 /** chars that must be escaped: syntax, C0 + C1 controls */
-const REGEX_SYNTAX_CHAR = /^[\u0000-\u001F\u007F-\u009F{}\[\]\\?.^$*()/-]$/;
+const REGEX_SYNTAX_CHAR = /^[\u0000-\u001F\u007F-\u009F{}\[\]\\?|.^$*()/-]$/;
```
Comment (mcdurdin) [inline common/web/types/src/util/util.ts:130]: 😲

**Review by mcdurdin — APPROVED** [review]
LGTM

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.297-beta](https://keyman.com/downloads/releases/beta/17.0.297)

**Pull request merged**